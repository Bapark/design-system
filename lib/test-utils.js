import TestRenderer from 'react-test-renderer';

/**
 * Create a wrapper object from which nodes can be found and assertions can be made.
 * @param {object} element - React element to create a rendered instance for.
 */
export function createWrapper(element) {
  return TestRenderer.create(element);
}

/**
 * Find a single descendant with the provided type. Throws an error if more than one descendants
 * match.
 * @param {object} wrapper - an instance as returned by `ReactTestRenderer.create`
 * @param {(string|function)} type - component constructor or string name for built-ins
 */
export function findByType(wrapper, type) {
  return wrapper.root.findByType(type);
}

/**
 * Get a specific prop from a rendered test component.
 * @param {object} instance - an instance returned by `findByType`
 * @param {string} prop - name of the prop to extract
 */
export function getProp(instance, prop) {
  return instance.props[prop];
}
