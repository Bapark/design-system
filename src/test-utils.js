/**
 * Find a single descendant with the provided type. Throws an error if more than one descendants
 * match.
 * @param {object} wrapper - an instance as returned by `ReactTestRenderer.create`
 * @param {(string|function)} type - component constructor or string name for built-ins
 */
export function findByType(wrapper, type) {
  return wrapper.root.findByType(type);
}

/**
 *
 * @param {object} instance - an instance returned by `findByType`
 * @param {string} prop - name of the prop to extract
 */
export function getProp(instance, prop) {
  return instance.props[prop];
}
