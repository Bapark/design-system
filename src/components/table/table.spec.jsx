import React from 'react';
import { Table, TableHead, TableBody, TableRow, TableCell } from './index';
import { createWrapper } from '../../test-utils';

function render() {
  return createWrapper(
    <Table>
      <TableHead>
        <TableCell>id</TableCell>
        <TableCell>name</TableCell>
      </TableHead>
      <TableBody>
        <TableRow>
          <TableCell>1</TableCell>
          <TableCell>Bob Ross</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>2</TableCell>
          <TableCell>Bob Marley</TableCell>
        </TableRow>
      </TableBody>
    </Table>,
  );
}

test('rendering', () => {
  const wrapper = render();
  expect(wrapper).toMatchSnapshot();
});
