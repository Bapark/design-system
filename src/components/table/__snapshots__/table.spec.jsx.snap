// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<table
  className="table"
>
  <thead
    className="head"
  >
    <tr>
      <th>
        id
      </th>
      <th>
        name
      </th>
    </tr>
  </thead>
  <tbody
    className="body"
  >
    <tr
      className="row"
    >
      <td>
        1
      </td>
      <td>
        Bob Ross
      </td>
    </tr>
    <tr
      className="row"
    >
      <td>
        2
      </td>
      <td>
        Bob Marley
      </td>
    </tr>
  </tbody>
</table>
`;
